core/src/main/scala/io/finch/Endpoints.scala

```diff
@@ -466,24 +466,52 @@ trait Endpoints {
   /**
    * Maintains Basic HTTP Auth for an arbitrary [[Endpoint]].
    */
-  case class BasicAuth(user: String, password: String) {
-    private[this] val userInfo = s"$user:$password"
-    private[this] val expected = "Basic " + Base64StringEncoder.encode(userInfo.getBytes)
-
+  case class BasicAuth(realm: String)(authenticate: (String, String) => Boolean) {
```

Comment:
I wonder if we need to support `(String, String) => Future[Boolean]` instead so we could, for example, call the database asynchronously.